gammapy/spectrum/simulation_realbkg.py

```diff
@@ -0,0 +1,220 @@
+
+        Parameters
+        ----------
+        obs_id : int
+            Observation identifier
+        seed : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+            see :func:~`gammapy.utils.random.get_random_state`
+        """
+        random_state = get_random_state(seed)
+        self.simulate_source_counts(random_state)
+        obs = SpectrumObservation(on_vector=self.on_vector,
+                                  off_vector=self.off_vector,
+                                  aeff=self.aeff,
+                                  edisp=self.edisp)
+        self.simulate_background_counts(bkg_res)
+        obs.off_vector=self.off_vector
+        obs.obs_id = obs_id
+        self.obs = obs
+
+    def simulate_source_counts(self, rand):
+        """Simulate source `~gammapy.spectrum.PHACountsSpectrum`.
+
+        Source counts are added to the on vector.
+
+        Parameters
+        ----------
+        rand: `~numpy.random.RandomState`
+            random state
+        """
+        on_counts = rand.poisson(self.npred_source.data.data.value)
+
+        on_vector = PHACountsSpectrum(energy_lo=self.e_reco.lower_bounds,
+                                      energy_hi=self.e_reco.upper_bounds,
+                                      data=on_counts,
+                                      backscal=1,
+                                      meta=self._get_meta())
+        on_vector.livetime = self.src_livetime
+        self.on_vector = on_vector
+
+    def estimate_reflected(self, EXCLUSION_FILE, size):
+
+        #just extracts the reflected background
+        on_size=0.11 * u.deg #0.11 for point source cuts...
+        allsky_mask = SkyImage.read(EXCLUSION_FILE)
+        exclusion_mask = allsky_mask.cutout(position=self.obspos,size=size)
+        on_region=CircleSkyRegion(self.obspos,on_size)         
+        background_estimator = ReflectedRegionsBackgroundEstimator(obs_list=self.obslist, on_region=on_region, exclusion_mask = exclusion_mask)
+
+        background_estimator.run()
+        return background_estimator.result[0]
+
+
+
+    def simulate_background_counts(self, bkg_res):
+
+        a_off=bkg_res.a_off
+        a_on = bkg_res.a_on
+        nOFF= len(bkg_res.off_events.table)
+        alpha=float(a_on)/float(a_off)
+        nbkg=alpha*nOFF # number of background events in on region
+        nbkg = np.random.poisson(nbkg)
+        idxON=np.random.choice(np.arange(nOFF),nbkg,replace=False)
+
+        bkg_ev=bkg_res.off_events.table[idxON] #background events in on region
+        bkg_hist,edge=np.histogram(bkg_ev["ENERGY"],self.on_vector.energy.bins)
+        
+        # Add background to on_vector
+        self.on_vector.data.data += bkg_hist * u.ct
+
+        # Create off vector
+        off_events=bkg_res.off_events.table
+        off_events.remove_rows(idxON)
+        alpha_new=float(a_on)/float(a_off - 1)
+
+        off_counts,edge=np.histogram(off_events["ENERGY"],self.on_vector.energy.bins)
```

Comment:
Here you're calling `np.histogram` on event energies.

I think we should try to limit ourselves to "binned" simulators by just calling `np.random.poisson`, in the same way how Sherpa only has the `sherpa.fake` calls that poisson-fluctuate counts from npred vectors.

Long-term we might also get unbinned analysis or simulators (i.e. event samplers) in Gammapy, but those should probably be added as separate new classes, not mixed in with the existing binned analysis / simulation code.

@AtreyeeS - Am I missing something here? Or is re-organising your code possible to allow your use case in this way, i.e. move this `np.histogram` call and event handling into an earlier step, where you do that first in your script and then use the existing binned event simulator?